### PR TITLE
Mobile - Recipient Registration form flow

### LINKED ISSUE #299: Mobile - Recipient Registration form flow
To do
- [ ] "Next" button should be on the right at all times
- [x] Scrolling to bottom should not lift nav bar
- [x] ProgressBar should be sticky to top
- [x] Edge case: long string bring placed in markdown field will render off screen
- [x] Placeholder for URLS changed from example to ex
- [x] Full width buttons in Upload
- [x] Funding address layout improvement
- [ ] address shortened for mobile when transaction submitted

### LINKED ISSUE #332: Refactor address display method into a utility
We currently have a few components concatenating wallet addresses to ABCD....DCBA. This should be abstracted out into a utility function.

### PR DESCRIPTION
**Pull request by corwintines**
Fixes: https://github.com/ethereum/clrfund/issues/299
Fixes: https://github.com/ethereum/clrfund/issues/332
Notion:  https://www.notion.so/efdn/Mobile-Recipient-Registration-form-flow-71d1a9f1afe54e409db0ca0da8c15427
Would check the Notion card for reference on requested changes

**What Changed**
- Refactor for rendering address and hashes for the application
- FormNavigation refactor and style update for mobile
- FormProgressWidget changes to allow it to stick to top when in mobile
- Markdown overflow handled
- Placeholder changed from example to ex
- Full width buttons in image upload

### CODE REVIEW

File: vue-app/src/utils/renderAddressOrHash.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export function renderAddressOrHash(
```
Comment (samajammin) [inline vue-app/src/utils/renderAddressOrHash.ts]: I don't believe this is needed anymore - @wackerow removed these in https://github.com/ethereum/clrfund/pull/314

Comment (samajammin) [reply vue-app/src/utils/renderAddressOrHash.ts]: Actually never mind looks like we still use it in one place: https://github.com/ethereum/clrfund/blob/develop/vue-app/src/components/TransactionReceipt.vue#L75

Comment (samajammin) [reply vue-app/src/utils/renderAddressOrHash.ts]: I think we should just put this function in utils/accounts.ts vs creating a new file:
https://github.com/ethereum/clrfund/blob/develop/vue-app/src/utils/accounts.ts

Comment (corwintines) [reply vue-app/src/utils/renderAddressOrHash.ts]: Ill move this method to the accounts file, and do a passover of where it will be used again.

**Comment by samajammin** [comment]
Something is broken with the FormNavigation.vue logic - the final submit step of the recipients flow now has a "finish" button:
clr fund - clr fund 2021-09-01 at 6 13 12 PM

Clicking it takes me here https://deploy-preview-321--eth2clrfund.netlify.app/#/join/submit
clr fund - clr fund 2021-09-01 at 6 13 20 PM

There should only be a "previous" button on this step.

**Comment by samajammin** [comment]
@corwintines we're you able to successfully submit a project on mobile w/ the preview deploy (https://deploy-preview-321--eth2clrfund.netlify.app/#/join/)? 

I got to the submit step but for some reason MetaMask mobile app disabled the button - I couldn't approve the tx:
76DAB28D-9D3B-40B1-92F2-985900B9C641_1_102_o
Not sure why - says "Unknown Method" & I can't click to "Confirm" 🤔

**Comment by samajammin** [comment]
Another weird situation - I tried rejecting the tx in MM mobile, which worked as expected:
787EA8BD-AB84-4568-AE7C-719F5272ABB1_1_102_o

At some point I got to a state where it displayed both that warning AND the waiting prompt:
6A4C7975-E0D3-4038-B47B-6E65C767B313_1_102_o
I don't think both of these prompts should ever display at the same time...

File: vue-app/src/components/FormNavigation.vue [hunk 2]
```diff
@@ -5,38 +5,18 @@
...
+        v-if="currentStep < steps.length - 1"
         @click="handleNext"
-        class="btn-primary"
+        class="btn-primary float-right"
         :disabled="!isStepValid"
       >
-        {{ currentStep === 5 ? 'Finish' : 'Next' }}
-      </button>
-    </div>
-    <div class="btn-row" v-if="!isJoin">
```
Comment (samajammin) [inline vue-app/src/components/FormNavigation.vue:22]: Note the `isJoin` prop is there to distinguish between the recipient flow & the BrightID flow. If we no longer need this prop we should remove it from the component & wherever it's used (but please check both flows to ensure the functionality still works as expected).

**Review by samajammin — CHANGES_REQUESTED** [review]
Noticed a few things while testing - see comments.

File: vue-app/src/components/FormNavigation.vue (hunk 2 above)
Comment (corwintines) [reply vue-app/src/components/FormNavigation.vue:22]: The FormNavigation is only getting used in the Join and FormProgressWidget (which is in Join) at the moment. Will remove the isJoin references as its not necessary now.

**Comment by corwintines** [comment]
@samajammin I added changes other than the for metamask portion. I found for that I was able to get it to work when adjusting gas. Thats not to say that isnt an issue, but I think this would maybe be better suited for some walletconnect polish. Let me know if you disagree.

**Review by samajammin — APPROVED** [review]
👍

**Pull request merged**